examples/netpkt/netpkt_ethercat.c

```diff
@@ -0,0 +1,114 @@
+/****************************************************************************
+ * apps/examples/netpkt/netpkt_ethercat.c
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.  The
+ * ASF licenses this file to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the
+ * License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ ****************************************************************************/
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <nuttx/config.h>
+
+#include <stdio.h>
+#include <time.h>
+
+#include <net/if.h>
+#include <netinet/ip.h>
+#include <netpacket/packet.h>
+#include <sys/socket.h>
+
+#include <nuttx/clock.h>
+
+/****************************************************************************
+ * Public Functions
+ ****************************************************************************/
+
+int main(int argc, FAR const char *argv[])
+{
+  struct timespec send_time;
+  struct timespec recv_time;
+  struct sockaddr_ll addr;
+  uint8_t sendbuff[60] =
+  {
+    0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0x9c, 0x7f, 0x81, 0x3c, 0x1e, 0x8e,
+    0x88, 0xa4, 0x0e, 0x10, 0x07, 0x4a, 0x00, 0x00, 0x30, 0x01, 0x02, 0x00,
+  };
+
+  uint8_t recvbuff[100];
+  FAR const char *ifname = "eth0";
+  int repeat = 10;
+  int num_packets = 0;
+  int len;
+  int ifindex;
+  int sockfd = socket(AF_PACKET, SOCK_RAW, IPPROTO_RAW);
+
+  if (sockfd == -1)
+    {
+      perror("ERROR: creating socket failed");
+      return -errno;
+    }
+
+  if (argc >= 2)
+    {
+      ifname = argv[1];
```

Comment:
assigning variable length input to const char[4] ?

Reply:
> assigning variable length input to const char[4] ?

Is it modified like this `int main(int argc, FAR const char *argv[4])`? Done.

Reply:
why make this change? all.main should havr the same prototype
int main(int argc, char *argv[]);